Modules/_io/iobase.c

```diff
@@ -671,6 +672,7 @@ _io__IOBase_readlines_impl(PyObject *self, Py_ssize_t hint)
 
     while (1) {
         PyObject *line = PyIter_Next(it);
+        Py_ssize_t line_length;
```

Comment:
There is no mixing variable declaration and assignment. `PyObject *line = PyIter_Next(it);` is not an assignment, it is an initialization. But I'll swap the lines because this makes `line` initialization closer to checking it's value.